src/process.js

```diff
@@ -264,14 +269,28 @@ class Process extends events.EventEmitter {
     }
   }
 
+  _isProbablyDeleted(item) {
+    // Can't really tell if a database has been deleted or just hasn't been
+    //  replicated to our node yet.  Compromise by deleting the item
+    //  if updated_at is long ago enough
+    return (
+      new Date().getTime() - new Date(item.updated_at).getTime() > this._assumeDeletedAfterSeconds * 1000
+    )
+  }
+
   async _processAndUnlockIfError(item) {
```

Comment:
I'd recommend adding a unit test for _processAndUnlockIfError in https://github.com/redgeoff/spiegel/blob/master/test/spec/process.js that stubs _process to throw a DatabaaseNotFoundError and then use a spy to make sure _getAndDestroy is called